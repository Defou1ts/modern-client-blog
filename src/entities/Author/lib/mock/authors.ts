import type { AuthorWithLocales } from '../../interfaces';

export const authors: AuthorWithLocales[] = [
	{
		id: 1,
		avatarURL:
			'https://images.pexels.com/photos/7562313/pexels-photo-7562313.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Floyd',
				surname: 'Miles',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Флойд',
				surname: 'Майлз',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 2,
		avatarURL:
			'https://images.pexels.com/photos/16424676/pexels-photo-16424676.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Diana',
				surname: 'Russel',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Дианал',
				surname: 'Рассл',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 3,
		avatarURL:
			'https://images.pexels.com/photos/5704849/pexels-photo-5704849.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Jenny',
				surname: 'Wilson',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Дженни',
				surname: 'Уилсон',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 4,
		avatarURL:
			'https://images.pexels.com/photos/6205509/pexels-photo-6205509.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Leslie',
				surname: 'Alexander',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Александр',
				surname: 'Лесли',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 5,
		avatarURL:
			'https://images.pexels.com/photos/14687031/pexels-photo-14687031.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Guy',
				surname: 'Hawkins',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Гай',
				surname: 'Хавкинс',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 6,
		avatarURL:
			'https://images.pexels.com/photos/4467687/pexels-photo-4467687.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Eleanor',
				surname: 'Pena',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Элеонора',
				surname: 'Пена',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 7,
		avatarURL:
			'https://images.pexels.com/photos/5262459/pexels-photo-5262459.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Robert',
				surname: 'Fox',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Роберт',
				surname: 'Фокс',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
	{
		id: 8,
		avatarURL:
			'https://images.pexels.com/photos/6147118/pexels-photo-6147118.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=1',
		content: {
			en: {
				name: 'Jacob',
				surname: 'Jones',
				position: 'Content Writer',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
			ru: {
				name: 'Джейкоб',
				surname: 'Джонс',
				position: 'Контент мейкер',
				company: 'Modsen',
				description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus.'
			},
		},
		socials: [
			{
				type: 'facebook',
				path: 'https://facebook.com/ModsenSoftware/',
			},
			{
				type: 'twitter',
				path: 'https://twitter.com/modsencompany',
			},
			{
				type: 'instagram',
				path: 'https://www.instagram.com/modsencompany/',
			},
			{
				type: 'linkedin',
				path: 'https://linkedin.com/company/modsen',
			},
		],
	},
];
