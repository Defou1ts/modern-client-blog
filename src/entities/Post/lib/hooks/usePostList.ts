import { useState, useMemo } from 'react';

import type { Post } from '../../interfaces';

export const usePostList = (posts: Post[], maxPosts: number) => {
	const [currentPage, setCurrentPage] = useState(0);

	const handlePrevClick = () => {
		setCurrentPage(currentPage - 1);
	};

	const handleNextClick = () => {
		setCurrentPage(currentPage + 1);
	};

	const isPrevDisabled = useMemo(() => currentPage === 0, [currentPage]);
	const isNextDisabled = useMemo(
		() => currentPage * maxPosts + maxPosts >= posts.length,
		[currentPage, maxPosts, posts]
	);
	const needButtons = !(isNextDisabled && isPrevDisabled);

	const postsToRender = posts.slice(currentPage * maxPosts, currentPage * maxPosts + maxPosts);

	return {
		postsToRender,
		handlePrevClick,
		handleNextClick,
		isPrevDisabled,
		isNextDisabled,
		needButtons,
	};
};
