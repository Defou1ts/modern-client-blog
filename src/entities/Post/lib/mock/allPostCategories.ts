import BusinessIcon from '@app/assets/icons/business.png';
import StartupIcon from '@app/assets/icons/startup.png';
import EconomyIcon from '@app/assets/icons/economy.png';
import TechnologyIcon from '@app/assets/icons/technology.png';

import type { PostCategory } from '@entities/Post/interfaces';

export const allPostCategories: PostCategory[] = [
	{
		category: 'business',
		icon: BusinessIcon,
		title: {
			en: 'Business',
			ru: 'Бизнес',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
	{
		category: 'startup',
		icon: StartupIcon,
		title: {
			en: 'Startup',
			ru: 'Стартап',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
	{
		category: 'economy',
		icon: EconomyIcon,
		title: {
			en: 'Economy',
			ru: 'Экономика',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
	{
		category: 'technology',
		icon: TechnologyIcon,
		title: {
			en: 'Technology',
			ru: 'Технологии',
		},
		description: {
			en: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
			ru: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
		},
	},
];
