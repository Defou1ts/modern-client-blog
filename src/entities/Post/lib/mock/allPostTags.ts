import type { PostTag } from '../../types';

export const allPostTags: PostTag[] = ['business', 'experience', 'screen', 'technology', 'marketing', 'life'];
