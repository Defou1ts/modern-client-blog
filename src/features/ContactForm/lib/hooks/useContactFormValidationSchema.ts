import { useTranslation } from 'next-i18next';
import * as Yup from 'yup';

import type { ContactFormState } from '../../interfaces';

export const useContactFormValidationSchema = () => {
	const { t } = useTranslation();

	const validationSchema = Yup.object<ContactFormState>({
		name: Yup.string().required(t('contact.form.errors.name') ?? ''),
		email: Yup.string().required(t('contact.form.errors.email') ?? ''),
		location: Yup.string().required(t('contact.form.errors.location') ?? ''),
		message: Yup.string()
			.required(t('contact.form.errors.requiredmessage') ?? '')
			.min(40, t('contact.form.errors.message') ?? ''),
	});

	return validationSchema;
};
