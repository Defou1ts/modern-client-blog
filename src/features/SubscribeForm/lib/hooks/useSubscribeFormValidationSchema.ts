import { useTranslation } from 'next-i18next';
import * as Yup from 'yup';

import type { SubscribeFormState } from '../../interfaces';

export const useSubscribeFormValidationSchema = () => {
	const { t } = useTranslation();

	const validationSchema = Yup.object<SubscribeFormState>({
		email: Yup.string().email(t('subscribe.error') ?? ''),
	});

	return validationSchema;
};
