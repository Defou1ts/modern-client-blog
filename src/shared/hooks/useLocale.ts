import { useRouter } from 'next/router';

import { defaultLocale } from '@shared/contants/defaultLocale';

export const useLocale = () => {
	const { locale } = useRouter();

	return { locale: locale ?? defaultLocale };
};
