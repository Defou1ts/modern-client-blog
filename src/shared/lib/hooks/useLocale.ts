import { useRouter } from 'next/router';

import { defaultLocale } from '@shared/lib/contants/defaultLocale';

export const useLocale = () => {
	const { locale } = useRouter();

	return { locale: locale ?? defaultLocale };
};
