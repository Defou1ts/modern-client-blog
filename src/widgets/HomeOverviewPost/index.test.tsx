import React from 'react';

import { render, screen } from '@testing-library/react';
import { useRouter } from 'next/router';

import renderer from 'react-test-renderer';

import { HomeOverviewPost } from './index';

import { useLocale } from '../../shared/lib/hooks/useLocale';

import { posts } from '@entities/Post/lib/mock/posts';
import { authors } from '@entities/Author/lib/mock/authors';

import type { NextRouter } from 'next/router';

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}));

jest.mock('../../shared/lib/hooks/useLocale', () => ({
	useLocale: jest.fn(),
}));

describe('HomeOverviewPost', () => {
	beforeAll(() => {
		const useRouterMock = {
			locale: 'en',
			locales: ['en', 'ru'],
			pathname: '/',
		};
		const useLocaleMock = {
			locale: 'en',
		};
		(useRouter as jest.Mock<NextRouter>).mockReturnValue(useRouterMock as NextRouter);
		(useLocale as jest.Mock).mockReturnValue(useLocaleMock);
	});
	it('renders a  HomeOverviewPost', () => {
		render(<HomeOverviewPost post={posts[0]} author={authors[0]} />);
		const text = screen.getByText('home.overview.posted-on');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('HomeOverviewPost snapshot', () => {
		const component = renderer.create(<HomeOverviewPost post={posts[0]} author={authors[0]} />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
