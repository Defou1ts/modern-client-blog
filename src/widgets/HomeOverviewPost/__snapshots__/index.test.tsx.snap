// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeOverviewPost HomeOverviewPost snapshot 1`] = `
<section
  className="wrapper"
  data-test-id="home-overview-post"
>
  <img
    alt="Step-by-step guide to choosing great font pairs"
    className="backgroundImage"
    data-nimg="1"
    decoding="async"
    height={40}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    title="Step-by-step guide to choosing great font pairs"
    width={40}
  />
  <div
    className="backgroundEffectWrapper"
  >
    <div
      className="content"
    >
      <p
        className="categoryInfo"
      >
        home.overview.posted-on
         
        <span
          className="category"
        >
          Startup
        </span>
      </p>
      <h1
        className="h display title"
      >
        Step-by-step guide to choosing great font pairs
      </h1>
      <p
        className="p medium author"
      >
        by
         
        <a
          className="yellow"
          href="/author/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Floyd Miles
        </a>
          
        | 
          
        April 25, 2023
      </p>
      <p
        className="p medium description"
      >
        Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
      </p>
      <button
        className="button primary button"
      >
        <a
          data-test-id="home-overview-post-link"
          href="/posts/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          home.overview.read-more
        </a>
      </button>
    </div>
  </div>
</section>
`;
